pgxn/neon/walproposer_pg.c

```diff
@@ -289,6 +291,15 @@ WalproposerShmemInit(void)
 	return found;
 }
 
+static void
+WalproposerShmemInit_SyncSafekeeper(void)
+{
+	walprop_shared = palloc(WalproposerShmemSize());
```

Comment:
Some confusion here: shared or shame in Postgres means that segment is placed in shared memory.
But here it is allocated using palloc: it is process private memory.

Reply:
Yes, and normally it is in shmem, but from what I understand `SyncSafekeeper` is only one process and it doesn't initialize shmem using the normal callbacks, so we just allocate it there.

Reply:
Generally it feels like `SyncSafekeeper` doesn't really need shmem because it's an alone process without need to share memory with anyone. Maybe it doesn't need to initialize shmem at all then?

Reply:
Discussed with Arthur, seems that SyncSafekeeper uses NeonWalReader to write WAL to safekeepers, so it needs to know about donor (which now lives in shmem). So we need to initialize shmem even in syncsafekeeper mode.